### ISSUE TITLE
`abbrevs` xontrib seems to be broken after update to latest version of `xonsh` (0.12.0)

### ISSUE THREAD

**Issue by taranlu-houzz** [xonsh/xonsh#4755]
## Xonsh Config
```
xontrib load abbrevs
abbrevs["hello"] = "echo hello"
```

## Expected Behavior
The `abbrevs` specified should work correctly as in previous versions. At first I though it was because I had some abbrevs that use `..` in them (e.g. `abbrevs[".."] = "cd .."`, since the traceback noted it was hitting recursion depth limit when splitting on `.`, but it seems to have issues with all abbrevs.

## Current Behavior
```
↴3 ~
❯ xonsh
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
RecursionError: maximum recursion depth exceeded while calling a Python object
error running xonsh run control file '/Users/----/.xonshrc': maximum recursion depth exceeded while calling a Python object
---- ~ $
```

### Traceback
```
Traceback (most recent call last):
  File "/Users/----/.xonshrc", line 21, in <module>
    abbrevs["hello"] = "echo hello"
  File "/Users/----/.local/pipx/venvs/xonsh/lib/python3.10/site-packages/xonsh/built_ins.py", line 722, in __setitem__
    return self.obj.__setitem__(item, value)
  File "/Users/----/.local/pipx/venvs/xonsh/lib/python3.10/site-packages/xonsh/built_ins.py", line 710, in __getattr__
    return getattr(self.obj, name)
  File "/Users/----/.local/pipx/venvs/xonsh/lib/python3.10/site-packages/xonsh/built_ins.py", line 710, in __getattr__
    return getattr(self.obj, name)
  File "/Users/----/.local/pipx/venvs/xonsh/lib/python3.10/site-packages/xonsh/built_ins.py", line 710, in __getattr__
    return getattr(self.obj, name)
  [Previous line repeated 985 more times]
  File "/Users/----/.local/pipx/venvs/xonsh/lib/python3.10/site-packages/xonsh/built_ins.py", line 703, in obj
    names = self.objname.split(".")
RecursionError: maximum recursion depth exceeded while calling a Python object
```

## Steps to Reproduce
Just use the config above and try to run `xonsh`.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by jnoortheen** [comment]
Yes I can reproduce the issue. There is a workaround until a fix is released, 

`__xonsh__.builtins.abbrevs` to access abbrevs. Or set it in builtins like this 

```
import builtins
builtins.abbrevs = __xonsh__.builtins.abbrevs
```

**Comment by jnoortheen** [comment]
fixed with 0.12.1
